perf-dashboard/deployment-latency/tests/CollectDeploymentLatencyTest.php

```diff
@@ -0,0 +1,158 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\PerfDashBoard;
+
+use Google\Cloud\BigQuery\BigQueryClient;
+
+class CollectDeploymentLatencyTest extends \PHPUnit_Framework_TestCase
+{
+    const DEPLOYMENT_MAX_RETRY = 5;
+    const PROJECT_ID = 'php-perf-dash';
+    const DATASET_ID = 'deployment_latency';
+    const TABLE_ID = 'flex_deployments';
+
+    /* @var BigQueryClient */
+    private static $bigQuery;
+
+    /**
+     * @var string phpVersion The PHP version in a form of Major.Minor
+     * @return string a path to the directory
+     */
+    private static function createApp($phpVersion)
+    {
+        $tempDir = tempnam(sys_get_temp_dir(), 'app');
+        unlink($tempDir);
+        mkdir($tempDir);
+        mkdir($tempDir . '/web');
+        copy(__DIR__ . '/files/app.yaml', $tempDir . '/app.yaml');
+        copy(__DIR__ . '/files/web/index.php', $tempDir . '/web/index.php');
+        chdir($tempDir);
+        self::execWithError(
+            'composer require google/cloud',
+            'composer require google/cloud'
+        );
+        self::execWithError(
+            'composer --ignore-platform-reqs require php:' . $phpVersion . '.*',
+            'composer require php'
+        );
+        return $tempDir;
+    }
+
+    public static function setUpBeforeClass()
+    {
+        self::execWithError(
+            'gcloud config set app/use_runtime_builders true',
+            'use-runitme-builder'
+        );
+    }
+
+    public function testDeploymentLatency()
+    {
+        $phpVersions = [
+            '5.6',
+            '7.0',
+            '7.1'
+        ];
+        $types = [
+            'xrt',
+            'builder'
+        ];
+        foreach ($phpVersions as $phpVersion) {
+            $dir = self::createApp($phpVersion);
+            chdir($dir);
+            foreach ($types as $type) {
+                $reportName = sprintf('%s-%s', $type, $phpVersion);
+                $failureCount = 0;
+                if ($type === 'xrt') {
+                    $command = 'gcloud -q app deploy';
+                } else {
+                    $command = 'gcloud -q beta app deploy';
+                }
+                $command .= ' --version ' . str_replace('.', '', $reportName);
+                $command .= ' --no-stop-previous-version --no-promote';
+                $before = microtime(true);
+                $latency = 0.0;
+                while ($failureCount < self::DEPLOYMENT_MAX_RETRY) {
+                    $ret = self::execWithResult($command);
+                    if ($ret === 0) {
+                        $latency = microtime(true) - $before;
+                        break;
+                    }
+                    $failureCount++;
+                }
+                self::$bigQuery = self::createBigQueryClient();
```

Comment:
Since we're fetching the access token directly as a workaround, the token will be expired for later bigquery calls if we do it in `setupBeforeClass`, although I haven't tested it.